### DISCUSSION TITLE
How to not expose private api key in budibase? How to not expose private values such as API key in public form?

### DISCUSSION THREAD

**Discussion by kincjf** [Budibase/budibase#12150] [Help] [answered]
hello
Starting with simple membership registration, I want to make budibase the front-end that communicates with the strapi backend.
How to not expose private api key in budibase? How to not expose private values such as API key in public form?

Is this a paid feature? How much do I have to pay to use this feature (private key)?

https://github.com/Budibase/budibase/discussions/3368   It seems like a question similar to mine exists in November 2021
https://docs.budibase.com/docs/environment-variables   Is "environment variable" the function I'm thinking of?

**Comment by ConorWebb96** [comment, marked as answer]
Hey @kincjf,

Yes this would come under `Environment variables`. This would allow you to use variables that only admins would have access to inside your rest queries. Its really useful for storing keys etc. The [price](https://budibase.com/pricing/) varies, it depends on how many users you have and whether you have selected annual payment or monthly payment. 

The Business plan is the one that would unlock this feature. Depending on the billing method selected it will either be $16 annually per month or $20 monthly per user.

I hope this helps.
